linux/net/rina/pff-ps-default.c

```diff
@@ -530,47 +374,113 @@ int pft_remove(struct pft *      instance,
                 pfte_destroy(tmp);
         }
 
-        mutex_unlock(&instance->write_lock);
+        mutex_unlock(&priv->write_lock);
+
+        return 0;
+}
+
+static bool default_is_empty(struct pff_ps * ps)
+{
+        struct pff_ps_priv * priv;
+        bool                 empty;
+
+        priv = (struct pff_ps_priv *) ps->priv;
+        if (!priv_is_ok(priv))
+                return false;
+
+        rcu_read_lock();
+        empty = list_empty(&priv->entries);
+        rcu_read_unlock();
+
+        return empty;
+}
+
+static void __pft_flush(struct pff_ps_priv * priv)
+{
+        struct pft_entry * pos, * next;
+
+        ASSERT(priv_is_ok(priv));
+
+        list_for_each_entry_safe(pos, next, &priv->entries, next) {
+                pfte_destroy(pos);
+        }
+}
+
+static int default_flush(struct pff_ps * ps)
+{
+        struct pff_ps_priv * priv;
+
+        priv = (struct pff_ps_priv *) ps->priv;
+        if (!priv_is_ok(priv))
+                return -1;
+
+        mutex_lock(&priv->write_lock);
```

Comment:
You cannot take a mutex lock inside an RCU critical section.
You are in an RCU critical section here because you hold the PFF policy-set.